modules/azure/vpn/vpn.tf

```diff
@@ -0,0 +1,40 @@
+resource "azurerm_public_ip" "public_ip" {
+  count                        = "${var.vpn_enabled}"
+  name                         = "${var.cluster_name}-vpn-public-ip"
+  location                     = "${var.location}"
+  resource_group_name          = "${var.resource_group_name}"
+  public_ip_address_allocation = "Dynamic"
+}
+
+resource "azurerm_virtual_network_gateway" "gateway" {
+  count               = "${var.vpn_enabled}"
+  name                = "${var.cluster_name}-vpn-gateway"
+  location            = "${var.location}"
+  resource_group_name = "${var.resource_group_name}"
+
+  type     = "Vpn"
+  vpn_type = "RouteBased"
+
+  active_active = false
+  enable_bgp    = false
+
+  # SKU details can be found here
+  # https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-about-vpngateways
+  sku = "VpnGw1"
```

Comment:
`Basic` only supports 10 tunnels, so use `VpnGw1` which supports up to 30 tunnels, because we will use it for BYOC guest clusters.